File: scratch/convert_to_utf8.py
import os

path = r"d:\_Site\ssotansum\ncd\admin\import_hdc.php"
try:
    with open(path, 'rb') as f:
        content = f.read()
    # Try decoding as tis-620 or windows-874
    try:
        decoded = content.decode('tis-620')
        print("Successfully decoded as TIS-620")
    except Exception as e1:
        decoded = content.decode('cp874')
        print("Successfully decoded as CP874")
    
    # Save backup
    with open(path + '.bak', 'wb') as f:
        f.write(content)
    # Write as UTF-8
    with open(path, 'w', encoding='utf-8', newline='') as f:
        f.write(decoded)
    print("Converted to UTF-8 and backup saved.")
except Exception as e:
    print("Error:", e)

File: scratch/convert_to_utf8.php
<?php
$file = __DIR__ . '/../admin/import_hdc.php';
$content = file_get_contents($file);
$encoding = mb_detect_encoding($content, ['UTF-8', 'TIS-620', 'SJIS', 'EUC-JP', 'ASCII', 'ISO-8859-11']);
echo "Detected encoding: " . $encoding . "\n";

if ($encoding !== 'UTF-8') {
    // If it's TIS-620 or ISO-8859-11 (Thai ANSI)
    $converted = mb_convert_encoding($content, 'UTF-8', 'TIS-620');
    // Save backup
    file_put_contents($file . '.bak', $content);
    // Write UTF-8
    file_put_contents($file, $converted);
    echo "Converted file to UTF-8 and saved backup to import_hdc.php.bak\n";
} else {
    echo "File is already UTF-8 or could not be converted\n";
}
